frontend/shared/src/utils/__tests__/date.utils.test.ts

```diff
@@ -39,4 +45,11 @@ describe('dates', () => {
       expect(diffMonthsResult).toBeLessThan(0);
     });
   });
+
+  describe('getCorrectEndDate', () => {
+    it('should be expected result', () => {
+      expect(getCorrectEndDate('31.12.2022', '1.1.2021')).toBe('31.12.2022');
```

Comment:
for 100% path coverage you could also add two more checks with invalid string values.

Reply:
Third case would be that both valeus are invalid, it seem to return the first value even if it's invalid. Perhaps it could return undefined or null instead. That's a minor thing anyway.

Reply:
This is now fixed